go/tasks/pluginmachinery/core/resource_manager.go

```diff
@@ -36,6 +36,17 @@ type ResourceRegistrar interface {
 // Resource Manager manages a single resource type, and each allocation is of size one
 type ResourceManager interface {
 	GetID() string
-	AllocateResource(ctx context.Context, namespace ResourceNamespace, allocationToken string) (AllocationStatus, error)
+	AllocateResource(ctx context.Context, namespace ResourceNamespace, allocationToken string, constraintsSpec ResourceConstraintsSpec) (AllocationStatus, error)
```

Comment:
isnt this a little weird in the API, where the plugin writer needs to provide the constraintspec?

Reply:
my preference is that for a task execution context some of these things are provided by the task handler (in propeller)

Reply:
I designed it this way because I want to be able to allow different cap settings for different service clusters (e.g., Qubole clusters). For example, the `default` Qubole cluster should cap any namespace at 70% of the quotas, but we do NOT want the same thing in the `priceoptimizeroffline` Qubole cluster because the entire quotas of that cluster are dedicated to one team only. 

Currently, the config item `ClusterConfig` resides in the plugin's config map. That's where I find most natural for these cluster-specific cap settings.